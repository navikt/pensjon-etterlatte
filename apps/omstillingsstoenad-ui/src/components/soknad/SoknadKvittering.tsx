import React, { useEffect } from 'react'
import { SkjemaGruppe } from '../felles/SkjemaGruppe'
import { Alert, BodyLong, BodyShort, Button, Link, Heading } from '@navikt/ds-react'
import { useTranslation } from 'react-i18next'
import { useLanguage } from '../../hooks/useLanguage'
import { NavigasjonsRadSection } from '../felles/StyledComponents'

const SoknadKvittering = () => {
    const { t } = useTranslation()
    useLanguage()

    useEffect(() => {
        window.history.pushState(null, document.title, window.location.href)
        window.addEventListener('popstate', () => {
            window.history.pushState(null, document.title, window.location.href)
        })
    }, [])

    return (
        <>
            <Heading className={'center'} size={'medium'} spacing={true}>
                {t('soeknadKvittering.tittel')}
            </Heading>

            <SkjemaGruppe>
                <Alert variant={'success'}>{t('soeknadKvittering.mottatt')}</Alert>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <Heading size={'small'}>{t('soeknadKvittering.seSaken.tittel')}</Heading>

                <BodyLong>
                    {t('soeknadKvittering.seSaken.informasjon.innhold1')}&nbsp;
                    <Link href={t('soeknadKvittering.seSaken.informasjon.lenkeMittNAV.href')}>
                        {t('soeknadKvittering.seSaken.informasjon.lenkeMittNAV.tekst')}
                    </Link>
                    &nbsp;
                    {t('soeknadKvittering.seSaken.informasjon.innhold2')}&nbsp;
                    <Link href={t('soeknadKvittering.seSaken.informasjon.lenkeSaksbehandlingstid.href')}>
                        {t('soeknadKvittering.seSaken.informasjon.lenkeSaksbehandlingstid.tekst')}
                    </Link>
                </BodyLong>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <Alert variant={'info'}>
                    <Heading size={'small'}>{t('soeknadKvittering.endring.tittel')}</Heading>

                    <BodyLong>{t('soeknadKvittering.endring.informasjon')}</BodyLong>

                    <ul>
                        <li>
                            <BodyShort>{t('soeknadKvittering.endring.endringsListe.sivilstatus')}</BodyShort>
                        </li>
                        <li>
                            <BodyShort>{t('soeknadKvittering.endring.endringsListe.inntekt')}</BodyShort>
                        </li>
                        <li>
                            <BodyShort>{t('soeknadKvittering.endring.endringsListe.bosted')}</BodyShort>
                        </li>
                    </ul>
                    <BodyLong>
                        {t('soeknadKvittering.endring.rettigheter.informasjon')}&nbsp;
                        <Link href={t('soeknadKvittering.endring.rettigheter.lenke.href')}>
                            {t('soeknadKvittering.endring.rettigheter.lenke.tekst')}
                        </Link>
                    </BodyLong>
                </Alert>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <Heading size={'small'}>{t('soeknadKvittering.andreStoenader.tittel')}</Heading>

                <BodyLong>{t('soeknadKvittering.andreStoenader.informasjon')}</BodyLong>

                <ul>
                    <li>
                        <BodyShort>{t('soeknadKvittering.andreStoenader.stoenadListe.skolepenger.tekst')}</BodyShort>
                    </li>
                    <li>
                        <BodyShort>{t('soeknadKvittering.andreStoenader.stoenadListe.barnetilsyn.tekst')}</BodyShort>
                    </li>
                    <li>
                        <BodyShort>{t('soeknadKvittering.andreStoenader.stoenadListe.barnetrygd.tekst')}</BodyShort>
                    </li>
                </ul>

                <BodyShort spacing={true}>
                    {t('soeknadKvittering.andreStoenader.andreInformasjon')}
                    <Link href={t('soeknadKvittering.andreStoenader.stoenadListe.tillegg.href')}>
                        {t('soeknadKvittering.andreStoenader.stoenadListe.tillegg.tekst')}
                    </Link>
                </BodyShort>

                <br />
            </SkjemaGruppe>

            <SkjemaGruppe>
                <NavigasjonsRadSection>
                    <Button
                        variant={'secondary'}
                        type={'button'}
                        onClick={() => (window.location.href = 'https://www.nav.no/omstillingsstonad')}
                    >
                        {t('soeknadKvittering.spoersmaal.knapp')}
                    </Button>
                </NavigasjonsRadSection>
            </SkjemaGruppe>
        </>
    )
}

export default SoknadKvittering
