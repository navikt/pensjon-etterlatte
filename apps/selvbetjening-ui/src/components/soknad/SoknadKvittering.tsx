import "./SoknadForside.scss";
import Veileder from "nav-frontend-veileder";
import ikon from "../../assets/ikoner/veileder.svg";
import { useSoknadContext } from "../../context/soknad/SoknadContext";
import { ActionTypes } from "../../context/soknad/soknad";
import React, { useEffect } from "react";
import { SkjemaGruppe } from "nav-frontend-skjema";
import { Alert, BodyLong, BodyShort, Button, Link, Heading } from "@navikt/ds-react";
import { useTranslation } from "react-i18next";
import { useLanguage } from "../../hooks/useLanguage";
const SoknadKvittering = () => {
    const { t } = useTranslation();
    useLanguage();
    const { dispatch } = useSoknadContext();

    useEffect(() => {
        dispatch({ type: ActionTypes.TILBAKESTILL });
    }, [dispatch])

    return (
        <div className={"forside"}>
            <SkjemaGruppe className={"center"}>
                <Veileder>
                    <img alt="veileder" src={ikon}/>
                </Veileder>
            </SkjemaGruppe>

            <SkjemaGruppe className={"center"}>
                <Heading size={"medium"} spacing={true}>{t("soeknadKvittering.tittel")}</Heading>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <Alert variant={"info"}>
                    {t("soeknadKvittering.kontakt")}
                </Alert>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <Heading size={"medium"}>{t("soeknadKvittering.endring.tittel")}</Heading>

                <BodyLong>
                    {t("soeknadKvittering.endring.informasjon")}
                </BodyLong>

                <ul>
                    <li>
                        <BodyShort>
                            {t("soeknadKvittering.endring.endringsListe.sivilstatus")}
                        </BodyShort>
                    </li>
                    <li>
                        <BodyShort>
                            {t("soeknadKvittering.endring.endringsListe.inntekt")}
                        </BodyShort>
                    </li>
                    <li>
                        <BodyShort>
                            {t("soeknadKvittering.endring.endringsListe.bosted")}
                        </BodyShort>
                    </li>
                </ul>

                <BodyLong>
                    {t("soeknadKvittering.endring.rettigheter.informasjon")}&nbsp;
                    <Link href={t("soeknadKvittering.endring.rettigheter.lenke.href")}>{t("soeknadKvittering.endring.rettigheter.lenke.tekst")}</Link>
                </BodyLong>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <Heading size={"medium"}>{t("soeknadKvittering.seSaken.tittel")}</Heading>

                {/* TODO: Sett inn riktig lenke for Dine saker */}
                <BodyLong>
                    {t("soeknadKvittering.seSaken.informasjon.innhold1")}&nbsp;
                    <Link href={t("soeknadKvittering.seSaken.informasjon.lenkeDittNAV.href")}>{t("soeknadKvittering.seSaken.informasjon.lenkeDittNAV.tekst")}</Link>&nbsp;
                    {t("soeknadKvittering.seSaken.informasjon.innhold2")}&nbsp;
                    <Link href={t("soeknadKvittering.seSaken.informasjon.lenkeDineSaker.href")}>{t("soeknadKvittering.seSaken.informasjon.lenkeDineSaker.tekst")}</Link>&nbsp;
                    {t("soeknadKvittering.seSaken.informasjon.innhold3")}&nbsp;
                </BodyLong>
                    <br/>
                <BodyLong>
                    <Link href={t("soeknadKvittering.seSaken.behandlingstidInfo.lenke.href")}>{t("soeknadKvittering.seSaken.behandlingstidInfo.lenke.tekst")}</Link>
                </BodyLong>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <Heading size={"medium"}>
                    {t("soeknadKvittering.andreStoenader.tittel")}
                </Heading>

                <BodyLong>
                    {t("soeknadKvittering.andreStoenader.informasjon")}
                </BodyLong>

                <ul>
                    <li>
                        <BodyShort>
                           <Link href={t("soeknadKvittering.andreStoenader.stoenadListe.skolepenger.href")}>
                               {t("soeknadKvittering.andreStoenader.stoenadListe.skolepenger.tekst")}
                           </Link>
                        </BodyShort>
                    </li>
                    <li>
                        <BodyShort>
                            <Link href={t("soeknadKvittering.andreStoenader.stoenadListe.barnetilsyn.href")}>
                                {t("soeknadKvittering.andreStoenader.stoenadListe.barnetilsyn.tekst")}
                            </Link>
                        </BodyShort>
                    </li>
                    <li>
                        <BodyShort>
                            <Link href={t("soeknadKvittering.andreStoenader.stoenadListe.barnetrygd.href")}>
                                {t("soeknadKvittering.andreStoenader.stoenadListe.barnetrygd.tekst")}
                            </Link>
                        </BodyShort>
                    </li>
                </ul>

                <br/>

                <BodyLong spacing={true}>
                    {t("soeknadKvittering.andreStoenader.andreInformasjon")}
                </BodyLong>

                <BodyShort>
                    <Link href={t("soeknadKvittering.andreStoenader.stoenadListe.tillegg.href")}>
                        {t("soeknadKvittering.andreStoenader.stoenadListe.tillegg.tekst")}
                    </Link>
                </BodyShort>

                <br/>
            </SkjemaGruppe>

            <SkjemaGruppe>
                <section className={"navigasjon-rad"}>
                    <Button
                        variant={"primary"}
                        type={"button"}
                        onClick={() => (window.location.href = "https://www.nav.no")}
                    >
                        {t("soeknadKvittering.spoersmaal.knapp")}
                    </Button>
                </section>
            </SkjemaGruppe>
        </div>
    );
};

export default SoknadKvittering;
