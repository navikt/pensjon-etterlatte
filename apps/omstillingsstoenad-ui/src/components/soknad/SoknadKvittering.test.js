import { render } from '@testing-library/react'
import SoknadKvittering from './SoknadKvittering'

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => ({ t: jest.fn((key) => key) }),
}))

describe('SoknadKvittering', () => {
    it('skal matche snapshot', () => {
        const { container } = render(<SoknadKvittering />)
        expect(container).toMatchSnapshot()
    })
})
