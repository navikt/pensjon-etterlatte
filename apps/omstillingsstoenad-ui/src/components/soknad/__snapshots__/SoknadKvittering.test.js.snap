// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoknadKvittering skal matche snapshot 1`] = `
<div>
  <h1
    class="center navds-heading navds-heading--medium navds-typo--spacing"
  >
    soeknadKvittering.tittel
  </h1>
  <fieldset
    class="sc-bdfCDU iGBHmN"
  >
    <div
      class="navds-alert navds-alert--success navds-alert--medium"
    >
      <svg
        aria-labelledby="title-r0"
        class="navds-alert__icon"
        fill="none"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="title-r0"
        >
          Suksess
        </title>
        <path
          clip-rule="evenodd"
          d="M12 21.75c5.385 0 9.75-4.365 9.75-9.75S17.385 2.25 12 2.25 2.25 6.615 2.25 12s4.365 9.75 9.75 9.75Zm4.954-12.475a.813.813 0 0 0-1.24-1.05l-5.389 6.368L7.7 11.967a.812.812 0 0 0-1.15 1.15l3.25 3.25a.812.812 0 0 0 1.195-.05l5.959-7.042Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="navds-alert__wrapper navds-body-long navds-body-long--medium"
      >
        soeknadKvittering.mottatt
      </div>
    </div>
  </fieldset>
  <fieldset
    class="sc-bdfCDU iGBHmN"
  >
    <h1
      class="navds-heading navds-heading--small"
    >
      soeknadKvittering.seSaken.tittel
    </h1>
    <p
      class="navds-body-long navds-body-long--medium"
    >
      soeknadKvittering.seSaken.informasjon.innhold1
       
      <a
        class="navds-link navds-link--action"
        href="soeknadKvittering.seSaken.informasjon.lenkeMittNAV.href"
      >
        soeknadKvittering.seSaken.informasjon.lenkeMittNAV.tekst
      </a>
       
      soeknadKvittering.seSaken.informasjon.innhold2
       
      <a
        class="navds-link navds-link--action"
        href="soeknadKvittering.seSaken.informasjon.lenkeSaksbehandlingstid.href"
      >
        soeknadKvittering.seSaken.informasjon.lenkeSaksbehandlingstid.tekst
      </a>
    </p>
  </fieldset>
  <fieldset
    class="sc-bdfCDU iGBHmN"
  >
    <div
      class="navds-alert navds-alert--info navds-alert--medium"
    >
      <svg
        aria-labelledby="title-r1"
        class="navds-alert__icon"
        fill="none"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title
          id="title-r1"
        >
          Informasjon
        </title>
        <path
          clip-rule="evenodd"
          d="M3.25 4A.75.75 0 0 1 4 3.25h16a.75.75 0 0 1 .75.75v16a.75.75 0 0 1-.75.75H4a.75.75 0 0 1-.75-.75V4ZM11 7.75a1 1 0 1 1 2 0 1 1 0 0 1-2 0Zm-1.25 3a.75.75 0 0 1 .75-.75H12a.75.75 0 0 1 .75.75v4.75h.75a.75.75 0 0 1 0 1.5h-3a.75.75 0 0 1 0-1.5h.75v-4h-.75a.75.75 0 0 1-.75-.75Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
      <div
        class="navds-alert__wrapper navds-body-long navds-body-long--medium"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          soeknadKvittering.endring.tittel
        </h1>
        <p
          class="navds-body-long navds-body-long--medium"
        >
          soeknadKvittering.endring.informasjon
        </p>
        <ul>
          <li>
            <p
              class="navds-body-short navds-body-short--medium"
            >
              soeknadKvittering.endring.endringsListe.sivilstatus
            </p>
          </li>
          <li>
            <p
              class="navds-body-short navds-body-short--medium"
            >
              soeknadKvittering.endring.endringsListe.inntekt
            </p>
          </li>
          <li>
            <p
              class="navds-body-short navds-body-short--medium"
            >
              soeknadKvittering.endring.endringsListe.bosted
            </p>
          </li>
        </ul>
        <p
          class="navds-body-long navds-body-long--medium"
        >
          soeknadKvittering.endring.rettigheter.informasjon
           
          <a
            class="navds-link navds-link--action"
            href="soeknadKvittering.endring.rettigheter.lenke.href"
          >
            soeknadKvittering.endring.rettigheter.lenke.tekst
          </a>
        </p>
      </div>
    </div>
  </fieldset>
  <fieldset
    class="sc-bdfCDU iGBHmN"
  >
    <h1
      class="navds-heading navds-heading--small"
    >
      soeknadKvittering.andreStoenader.tittel
    </h1>
    <p
      class="navds-body-long navds-body-long--medium"
    >
      soeknadKvittering.andreStoenader.informasjon
    </p>
    <ul>
      <li>
        <p
          class="navds-body-short navds-body-short--medium"
        >
          soeknadKvittering.andreStoenader.stoenadListe.skolepenger.tekst
        </p>
      </li>
      <li>
        <p
          class="navds-body-short navds-body-short--medium"
        >
          soeknadKvittering.andreStoenader.stoenadListe.barnetilsyn.tekst
        </p>
      </li>
      <li>
        <p
          class="navds-body-short navds-body-short--medium"
        >
          soeknadKvittering.andreStoenader.stoenadListe.barnetrygd.tekst
        </p>
      </li>
    </ul>
    <p
      class="navds-body-short navds-body-short--medium navds-typo--spacing"
    >
      soeknadKvittering.andreStoenader.andreInformasjon
      <a
        class="navds-link navds-link--action"
        href="soeknadKvittering.andreStoenader.stoenadListe.tillegg.href"
      >
        soeknadKvittering.andreStoenader.stoenadListe.tillegg.tekst
      </a>
    </p>
    <br />
  </fieldset>
  <fieldset
    class="sc-bdfCDU iGBHmN"
  >
    <section
      class="sc-hKgKIp jZdyrG"
    >
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        type="button"
      >
        <span
          class="navds-label"
        >
          soeknadKvittering.spoersmaal.knapp
        </span>
      </button>
    </section>
  </fieldset>
</div>
`;
